DashAI/back/dataloaders/classes/dataloader.py

```diff
@@ -0,0 +1,139 @@
+import io
+import zipfile
+from abc import abstractmethod
+
+from datasets import DatasetDict
+from fastapi import UploadFile
+
+from DashAI.back.config_object import ConfigObject
+
+
+class BaseDataLoader(ConfigObject):
+    """
+    Abstract class with base methods for all data loaders
+    """
+
+    @abstractmethod
+    def load_data(self, dataset_path, file=None, url=None):
+        raise NotImplementedError
+
+    def extract_files(self, dataset_path: str, file: UploadFile) -> str:
+        """
+        Extract the files to load the data in a DataDict later.
+
+        Args:
+            dataset_path (str): Path where dataset will be saved.
+            file (UploadFile): File uploaded for the user.
+
+        Returns:
+            str: Path of the files extracted.
+        """
+        if file.content_type == "application/zip":
+            files_path = f"{dataset_path}/files"
+            with zipfile.ZipFile(io.BytesIO(file.file.read()), "r") as zip_file:
+                zip_file.extractall(path=files_path)
+        else:
+            files_path = f"{dataset_path}/{file.filename}"
+            with open(files_path, "wb") as f:
+                f.write(file.file.read())
+        return files_path
+
+    def split_dataset(
+        self,
+        dataset: DatasetDict,
+        train_size: float,
+        test_size: float,
+        val_size: float,
+        seed: int = None,
+        shuffle: bool = True,
+        stratify: bool = False,
+        class_column: str = None,
+    ) -> DatasetDict:
+        """
+        Split the dataset in train, test and validation data.
+
+        Args:
```

Comment:
Add the split process documentation to the function docstring.
@cristian-tamblay , I copied and pasted @Kazymila 's comment, but I guess it needs to be better adapted.

```suggestion
        Split the dataset in train, test and validation data.

        First the dataset is splitting into a train split and a test-validation split, this is because the validation split is taken like a portion of the test split of the first splitting. For that the size of the test split for the first splitting is the sum of the sizes of test and validation splits.

        Then, the test and validation splits are defined in the second splitting, where now the train size of this splitting is the final test split and the result of the test split is actually the validation split. So that ‘val_size’ is the proportion of the validation split of the rest of the data in the resulting test split form first splitting process.

        An example, if I have a train size of 0.8, a test size of 0.1 and a validation size of 0.1, in the first process we split the data in the train data with the 80% of the data, and a test-validation data with the 20% remaining. Then in then second process we divide this 20% in a 10% of test data and 10% of validation data.

        Args:
```